### DISCUSSION TITLE
Posssible bug - TDR shows S11 even if data taken from S22

### DISCUSSION THREAD

**Discussion by davidkirkby** [scikit-rf/scikit-rf#920] [Q&A]
A graph plotted with TDR shows S11, even if the data is taken from S22 of a .s2p file. See the attached, which is basically taken from the manual. 

![S11-always-shows-even-if-from-s22](https://github.com/scikit-rf/scikit-rf/assets/124881781/9193b32f-7acf-4eef-8629-b0b7f0660737)

```
import skrf as rf
from pylab import *

# load data for the waveguide to CPW probe
probe = rf.Network('/tmp/probe.s2p')

# we will focus on s22 to demonstrate a problem. 
s22 = probe.s22

#  time-gate the first largest reflection
s22_gated = s22.time_gate(center=0, span=.2, t_unit='ns')
s22_gated.name='gated probe'

# plot frequency and time-domain s-parameters
figure(figsize=(8,4))
subplot(121)
s22.plot_s_db()
s22_gated.plot_s_db()
title('Frequency Domain')

subplot(122)
s22.plot_s_db_time()
s22_gated.plot_s_db_time()
title('Time Domain')
tight_layout()

plt.show()
```

**Comment by denzchoe** [comment]
Hiya, 

Depends on (how you read it) where is your reference point in the code,
You passed a 1 Network Obj to a variable here.

`s22 = probe.s22`

Therefore when it is plotted, it became a 1 network Port file, since you have already discarded your 2 port information.
<br>

According to the [documentation;](https://scikit-rf.readthedocs.io/en/latest/api/generated/skrf.network.Network.step_response.html#skrf.network.Network.step_response) <-- _I think there is a bug on this page. It is not displaying as it should be_
If you use the below script, you should be able to plot your S22 time domain information, while **retaining your 2 Port Network information** and displaying s22 in the Legend too.

`probe.plot_s_db_time(1,1)`
`probe.plot_s_db(1,1)`




<br>


The other way is you overwrite the legend, and you rename the legend manually (cumbersome but sometimes most efficient depending on the situation)

**Reply by davidkirkby** [reply]
Thank you, I will try later when I have access to a proper computer, not just a mobile phone.

However, on another graph (not shown), I plotted VSWR at ports 1 & 2. The legends then said something like

"VSWR port 1, S11"
"VSWR port 2, S11"

The second S11 made even less sense there, but it got automatically added. 

As I wrote in a previous post, I am pretty new to Python. I would actually like to overwrite most of the legends on the graphs, as I want to replace some words by French, despite I am English.

**Reply by denzchoe** [reply]
@davidkirkby ,
No problem man
This is how I usually do it... 

```python
probe.s11.plot_s_db(label = probe.name + ', any other string you want printed')
```
